pvlib/clearsky.py

```diff
@@ -644,8 +644,38 @@ def _clear_sample_index(clear_windows, samples_per_window, align, H):
     return clear_samples
 
 
-def detect_clearsky(measured, clearsky, times=None, window_length=10,
-                    mean_diff=75, max_diff=75,
+def _clearsky_get_threshold(sample_interval):
+    """
+    Returns threshold values for kwargs in detect_clearsky. See
+    Table 1 in [1].
+
+    References
+    ----------
+    .. [1] Jordan, D.C. and C. Hansen, "Clear-sky detection for PV
+       degradation analysis using multiple regression", Renewable Energy,
+       v209, p. 393-400, 2023.
+    """
+
+    if (sample_interval < 1 or sample_interval > 30):
+        raise ValueError("sample_interval must be larger than 1 and \
+                          smaller than 30")
```

Comment:
`sample_interval` is an internal variable in `detect_clearsky`, so the user won't necessarily know what this means.  Maybe reword to something like this?

> `infer_limits=True` can only be used for inputs with time step from 1 to 30 minutes